File: app/src/main/java/com/example/jetpackdemo/viewmodel/MyViewModel.java
package com.example.jetpackdemo.viewmodel;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class MyViewModel extends ViewModel {

    private MutableLiveData<Integer> likedNumber;
    public MutableLiveData<Integer> getLikedNumber(){
        if (likedNumber==null){
            likedNumber = new MutableLiveData<>();
            likedNumber.setValue(0);
        }
        return likedNumber;
    }

    public void setLikedNumber(int number){
        likedNumber.setValue(likedNumber.getValue()-number);
    }
}

File: app/src/main/java/com/example/jetpackdemo/day12_kotlindemo/MyViewModel.kt
package com.example.jetpackdemo.day12_kotlindemo

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class MyViewModel: ViewModel() {

    val number:MutableLiveData<Int> by lazy { MutableLiveData<Int>().also { it.value = 0 } }

    fun modifyNumber(aNumber:Int){
        number.value = number.value?.plus(aNumber)
    }
}